File: Javascript/Array/E-703. Kth Largest Element in a Stream.ts
class KthLargest {
  sortedNums: number[] = [];
  k: number;
  constructor(k: number, nums: number[]) {
    this.sortedNums = nums;
    this.k = k;
  }

  add(val: number): number {
    this.sortedNums.push(val);
    this.sortedNums.sort((a, b) => b - a);
    return this.sortedNums[this.k - 1];
  }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * var obj = new KthLargest(k, nums)
 * var param_1 = obj.add(val)
 */

File: Python/Array/E-703. Kth Largest Element in a Stream.py
class KthLargest:

    def __init__(self, k: int, nums: list[int]):
        self.k = k
        self.sorted_arr = nums

    def add(self, val: int) -> int:
        self.sorted_arr.append(val)
        self.sorted_arr.sort(reverse=True)
        return self.sorted_arr[self.k - 1]

        # Your KthLargest object will be instantiated and called as such:
        # obj = KthLargest(k, nums)
        # param_1 = obj.add(val)
